src/GafferScene/BranchCreator.cpp

```diff
@@ -115,22 +157,38 @@ class BranchCreator::BranchesData : public IECore::Data
 			// When multiple sources map to the same destination, we have no guarantees about
 			// the order they will be visited above. Sort them alphabetically so that our output
 			// is stable from run to run.
-			for( auto &it : m_sourcePaths )
-			{
-				std::sort(
-					it.second.begin(), it.second.end(),
-					[] ( const ScenePlug::ScenePath &a, const ScenePlug::ScenePath &b ) {
-						// InternedString compares by pointer address by default, so we need
-						// to manually compare by string contents.
-						return lexicographical_compare(
-							a.begin(), a.end(), b.begin(), b.end(),
+			visitLocationsWalk(
+				[] ( const ScenePlug::ScenePath &path, Location *location ) {
+					if( location->sourcePaths )
+					{
+						std::sort(
+							location->sourcePaths->begin(), location->sourcePaths->end(),
+							[] ( const ScenePlug::ScenePath &a, const ScenePlug::ScenePath &b ) {
+								// InternedString compares by pointer address by default, so we need
+								// to manually compare by string contents.
+								return lexicographical_compare(
+									a.begin(), a.end(), b.begin(), b.end(),
+									[] ( const InternedString &a, const InternedString &b ) {
+										return a.string() < b.string();
+									}
+								);
+							}
+						);
+					}
+					if( location->newChildNames )
+					{
+						std::sort(
```

Comment:
I do kinda wonder if it's even necessary to store newChildNames during the first traverse, and sort it here, vs just traversing the sorted location->children and putting in the key whenever location->exists is false.  Oh, of course location->children is sorted in the wrong order, so we would still need this sort ... but it could still be worthwhile?  If we're not doing a granular mutex on the first traverse, then moving work out if it could be pretty valuable?

Reply:
This seems like a pretty reasonable idea, but without some profiling showing that we have a problem I'm pretty reluctant to add any more complexity or churn.

Reply:
Talking to you this morning, you thought that this was worth changing after all.

Reply:
I'm not convinced by this any more. Longer term I expect the per-location mutex will make a lot of sense, in which case don't we want to keep things as they are, rather than move more work to the serial final stage? I don't think there's much point in shuffling this around until we've got a motivating performance test we can use to judge the results. And I suspect that typical destination maps are small enough that we won't find such a case for a while (I haven't seen the previous internal FilterResults in profiles, and we're not doing all that much more work here now).

Reply:
OK, fair enough.